### PR TITLE
Modify workflow to add sObjectsDefinitions test in redhat

### PR DESCRIPTION
**Pull request by CristiCanizales**
### What does this PR do?
- Adds sObjectsDefinitions to run in redhat

### What issues does this PR fix or reference?
@W-17044345@

### CODE REVIEW

**Review by mingxuanzhangsfdx — APPROVED** [review]

File: .github/workflows/coreE2EredHat.yml [hunk 1]
```diff
@@ -144,26 +144,29 @@ jobs:
   #   secrets: inherit
   #   with:
   #     automationBranch: ${{ inputs.automationBranch }}
+  # automationRepo: 'salesforcedx-vscode-automation-tests-redhat'
```
Comment (daphne-sfdc) [inline .github/workflows/coreE2EredHat.yml:147]: Need the same indent as `automationBranch`

File: .github/workflows/coreE2EredHat.yml [hunk 2]
```diff
@@ -144,26 +144,29 @@ jobs:
...
+      vscodeVersion: ${{ inputs.vscodeVersion }}
+      runId: ${{ inputs.runId }}
 
   # templates:
   #   if: ${{ inputs.templates }}
   #   uses: ./.github/workflows/runE2ETest.yml
   #   secrets: inherit
   #   with:
   #     automationBranch: ${{ inputs.automationBranch }}
+  # automationRepo: 'salesforcedx-vscode-automation-tests-redhat'
```
Comment (daphne-sfdc) [inline .github/workflows/coreE2EredHat.yml:169]: Same here about the indent

File: .github/workflows/coreE2EredHat.yml
Comment (daphne-sfdc) [inline .github/workflows/coreE2EredHat.yml:1]: There should be an `automationRepo` entry for `authentication` too - I see you added the entry to `miscellaneous` and `templates`, which are both also commented out.

**Review by daphne-sfdc — APPROVED** [review]
Approved! 🎉 

Please address the below nits before merging.

**Pull request merged**